src/AppInstallerCommonCore/Fonts.cpp

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include <dwrite_3.h>
+#include <AppInstallerStrings.h>
+#include <winget/Fonts.h>
+
+namespace AppInstaller::Fonts
+{
+    namespace
+    {
+        std::vector<std::filesystem::path> GetFontFilePaths(const wil::com_ptr<IDWriteFontFace>& fontFace)
+        {
+            UINT32 fileCount = 0;
+            THROW_IF_FAILED(fontFace->GetFiles(&fileCount, nullptr));
+
+            wil::com_ptr<IDWriteFontFile> fontFiles[8];
+            THROW_HR_IF(E_OUTOFMEMORY, fileCount > ARRAYSIZE(fontFiles));
+            THROW_IF_FAILED(fontFace->GetFiles(&fileCount, fontFiles[0].addressof()));
+
+            std::vector<std::filesystem::path> filePaths;
+            for (UINT32 i = 0; i < fileCount; ++i) {
+                wil::com_ptr<IDWriteFontFileLoader> loader;
+                THROW_IF_FAILED(fontFiles[i]->GetLoader(loader.addressof()));
+
+                const void* fontFileReferenceKey;
+                UINT32 fontFileReferenceKeySize;
+                THROW_IF_FAILED(fontFiles[i]->GetReferenceKey(&fontFileReferenceKey, &fontFileReferenceKeySize));
+
+                if (const auto localLoader = loader.try_query<IDWriteLocalFontFileLoader>()) {
+                    UINT32 pathLength;
+                    THROW_IF_FAILED(localLoader->GetFilePathLengthFromKey(fontFileReferenceKey, fontFileReferenceKeySize, &pathLength));
+                    pathLength += 1; // Account for the trailing null terminator during allocation.
+
+                    wchar_t path[MAX_PATH];
```

Comment:
You have a size already, allocate some memory (preferably directly in a string so that you don't have to copy it).